src/services/AbortRequestProcessor.ts

```diff
@@ -0,0 +1,93 @@
+import { EnvironmentVariables } from "./EnvironmentVariables";
+import { ServicesEnum } from "../models/enums/ServicesEnum";
+import { MessageCodes } from "../models/enums/MessageCodes";
+import { AuthSessionState } from "../models/enums/AuthSessionState";
+
+export class AbortRequestProcessor {
+
+  private static instance: AbortRequestProcessor;
+
+  private readonly logger: Logger;
+
+  private readonly metrics: Metrics;
+
+  private readonly f2fService: F2fService;
+
+  private readonly environmentVariables: EnvironmentVariables;
+
+  constructor(logger: Logger, metrics: Metrics) {
+  	this.logger = logger;
+  	this.metrics = metrics;
+  	this.environmentVariables = new EnvironmentVariables(logger, ServicesEnum.ABORT_SERVICE);
+  	this.f2fService = F2fService.getInstance(this.environmentVariables.sessionTable(), this.logger, createDynamoDbClient());
+  }
+
+  static getInstance(
+  	logger: Logger,
+  	metrics: Metrics,
+  ): AbortRequestProcessor {
+  	if (!AbortRequestProcessor.instance) {
+  		AbortRequestProcessor.instance =
+        new AbortRequestProcessor(logger, metrics);
+  	}
+  	return AbortRequestProcessor.instance;
+  }
+
+  async processRequest(sessionId: string): Promise<Response> {
+  	const f2fSessionInfo = await this.f2fService.getSessionById(sessionId);
+  	this.logger.appendKeys({
+  		govuk_signin_journey_id: f2fSessionInfo?.clientSessionId,
+  	});
+
+  	if (!f2fSessionInfo) {
+  		this.logger.warn("Missing details in SESSION TABLE", {
+  			messageCode: MessageCodes.SESSION_NOT_FOUND,
+  		});
+  		throw new AppError(HttpCodesEnum.BAD_REQUEST, "Missing details in SESSION table");
+  	}
+
+
+  	if (f2fSessionInfo.authSessionState === AuthSessionState.F2F_CRI_SESSION_ABORTED) {
+  		this.logger.info("Session has already been aborted");
+  	  return new Response(HttpCodesEnum.OK, "Session has already been aborted");
+  	}
+
+  	try {
+  	  await this.f2fService.updateSessionAuthState(f2fSessionInfo.sessionId, AuthSessionState.F2F_CRI_SESSION_ABORTED);
+  	} catch (error) {
+  		this.logger.error("Error occurred while aborting the session", {
+  			error,
+  			messageCode: MessageCodes.SERVER_ERROR,
+  		});
+  		if (error instanceof AppError) {
+  			return new Response(HttpCodesEnum.SERVER_ERROR, error.message);
+  		} else {
+  			return new Response(HttpCodesEnum.SERVER_ERROR, "An error has occurred");
+  		}
+  	}
+
+  	try {
+  		await this.f2fService.sendToTXMA({
+  			event_name: "F2F_CRI_SESSION_ABORTED",
+  			...buildCoreEventFields(f2fSessionInfo, this.environmentVariables.issuer() as string, f2fSessionInfo.clientIpAddress, absoluteTimeNow),
+  		});
+  	} catch (error) {
+  		this.logger.error("Auth session successfully aborted. Failed to send F2F_CRI_SESSION_ABORTED event to TXMA", {
+  			error,
+  			messageCode: MessageCodes.FAILED_TO_WRITE_TXMA,
+  		});
+  	}
+
+  	const redirectUri = `${f2fSessionInfo.redirectUri}?error=access_denied&state=${AuthSessionState.F2F_CRI_SESSION_ABORTED}`;
```

Comment:
I think `state` needs to be one from the session object, could you confirm please?

Reply:
@richa-misra do you mean the state before or after we've updated it to be `F2F_CRI_SESSION_ABORTED`?

Reply:
the one we are returning in back in the Response. We have `authSessionState` which maps to one of the values in src/models/enums/AuthSessionState.ts and `state` which is present in the incoming JWT from IPV. I think its the one from IPV that needs to be passed back.

Reply:
All done